WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostListViewModel.kt

```diff
@@ -249,6 +261,37 @@ class ReaderPostListViewModel @Inject constructor(
                 ))
     }
 
+    /**
+     * Handles reblog button action
+     *
+     * @param post post to reblog
+     */
+    fun onReblogButtonClicked(post: ReaderPost) {
+        val selectedSiteId = appPrefsWrapper.getSelectedSite()
+        val selectedSite: SiteModel = siteStore.getSiteByLocalId(selectedSiteId)
+        when (siteStore.visibleSites.size) {
+            0 -> _reblogAction.value = Event(NoSite)
+            1 -> _reblogAction.value = Event(PostEditor(selectedSite, post))
+            else -> _reblogAction.value = Event(SitePicker(selectedSite, post))
+        }
+    }
+
+    /**
+     * Handles site selection
+     *
+     * @param site selected site to reblog to
+     */
+    fun selectedSiteToReblog(siteLocalId: Int) {
+        val currentState = _reblogAction.value?.peekContent()
+        val selectedPost = currentState?.post
+        if (currentState is SitePicker && selectedPost != null) {
+            val site = siteStore.getSiteByLocalId(siteLocalId)
+            _reblogAction.value = Event(PostEditor(site, selectedPost))
+        } else {
+            _reblogAction.value = Event(ReblogError)
```

Comment:
Good to see you thought to some error handling here 👍! In cases like this I think we are dealing with some condition we do not expect (possibly a bug), so maybe worth to enforce fail fast at least for Debug configurations adding also something like this

```
                if (BuildConfig.DEBUG) {
                    throw IllegalStateException("Error Message")
                } else {
                    AppLog.e(T.READER, "Error Message")
                }
```

Wdyt? Let me know if I got your general intention wrong of course 😊.

Reply:
Fixed with 490ae8aab850f4f0e2f9c1e6c38b54d8a80eac07